frontend/exporter/src/constant_utils.rs

```diff
@@ -283,3 +407,104 @@ pub(crate) fn valtree_to_constant_expr<'tcx, S: BaseState<'tcx>>(
         attributes: vec![],
     }
 }
+
+pub(crate) fn const_value_reference_to_constant_expr<'tcx, S: BaseState<'tcx>>(
+    s: &S,
+    ty: rustc_middle::ty::Ty<'tcx>,
+    val: rustc_middle::mir::interpret::ConstValue<'tcx>,
+    span: rustc_span::Span,
+) -> ConstantExpr {
+    use rustc_middle::mir::interpret;
+    use rustc_middle::ty;
+
+    let tcx = s.base().tcx;
+
+    // We use [try_destructure_mir_constant] to destructure the constant
+    let param_env = get_param_env(s);
+    // We have to clone some values: it is a bit annoying, but I don't
+    // manage to get the lifetimes working otherwise...
+    let cvalue = rustc_middle::mir::ConstantKind::Val(val, ty);
+    let param_env_and_const = rustc_middle::ty::ParamEnvAnd {
+        param_env,
+        value: cvalue,
+    };
+
+    let dc = tcx
+        .try_destructure_mir_constant(param_env_and_const)
+        .unwrap();
+
+    // Iterate over the fields, which should be values
+    assert!(dc.variant.is_none());
+
+    // The type should be tuple
+    let hax_ty = ty.sinto(s);
+    match &hax_ty {
+        Ty::Tuple(_) => (),
+        _ => {
+            span_fatal!(s, span, "Expected the type to be tuple: {:?}", val)
+        }
+    };
```

Comment:
Why`hax_ty` and not `ty`? It's not used anywhere below, so we can shadow.
Also, I think using a `matches` would simply things
```suggestion
    let hax_ty = ty.sinto(s);
    if !matches(Ty::Tuple(_), &hax_ty) {
            span_fatal!(s, span, "Expected the type to be tuple: {:?}", val)
    };
```

Reply:
You can change that. For the `hax_ty`: the code was ported from Charon, where I often write `rust_...`